core/types/serialize/encode.go

```diff
@@ -112,11 +113,13 @@ func DecodeInto(bts SerializedData, v any) error {
 	return codec.Decode(val, v)
 }
 
-// Decode decodes the given serialized data into the given value.
-// See also DecodeInto for use with an existing instance.
-func Decode[T any](bts SerializedData) (*T, error) {
+// DecodeGeneric decodes the given serialized data into the given value. See
+// also Decode for use with an existing instance. This is generally no more
+// useful than Decode; it is syntactic sugar that requires no existing instance
+// of the type.
+func DecodeGeneric[T any](bts SerializedData) (*T, error) {
```

Comment:
I guess I'll remove this as discussed rather than rename.  It's hard to see a reason to use it imo